.circleci/config.yml

```diff
@@ -9,89 +12,87 @@ executors:
     steps:
       - checkout
-      - restore_cache:
-          key: sceptre-v1-{{ checksum "pyproject.toml" }}-{{ checksum "poetry.lock" }}-{{ checksum ".pre-commit-config.yaml" }}-{{ checksum "tox.ini" }}
       - add_ssh_keys:
           fingerprints:
-            - '80:24:c0:40:28:36:f1:80:7f:fd:15:c3:09:cf:86:fc'
+            - "80:24:c0:40:28:36:f1:80:7f:fd:15:c3:09:cf:86:fc"
       - attach_workspace:
           at: /home/circleci
+      - python/install-packages:
+          pkg-manager: poetry
+          args: --all-extras
       - run:
-          name: 'Build and deploy docs'
+          name: Build and deploy docs
           command: |
-            chmod +x .circleci/add-known-hosts.sh
             poetry run ./.circleci/add-known-hosts.sh
-            chmod +x .circleci/github-pages.sh
             poetry run ./.circleci/github-pages.sh
+          environment:
+            REPOSITORY_PATH: /home/circleci/docs
+            DEPLOYMENT_GIT_SSH: git@github.com:Sceptre/sceptre.github.io.git
 
 jobs:
   build:
-    docker:
-      - image: sceptreorg/sceptre-circleci:2.1.0
+    executor: python/default
     steps:
       - checkout
       - restore_cache:
-          key: sceptre-v1-{{ checksum "pyproject.toml" }}-{{ checksum "poetry.lock" }}-{{ checksum ".pre-commit-config.yaml" }}-{{ checksum "tox.ini" }}
-      - run:
-          name: 'Installing Dependencies'
-          command: poetry install --all-extras -v
+          key: sceptre-v1-{{ checksum ".pre-commit-config.yaml" }}
+      - python/install-packages:
+          pkg-manager: poetry
+          args: --all-extras
       - run:
           name: Run pre-commit
           command: poetry run pre-commit run --all-files --show-diff-on-failure
       - run:
           name: Build package
           command: poetry build
       - save_cache:
-          key: sceptre-v1-{{ checksum "pyproject.toml" }}-{{ checksum "poetry.lock" }}-{{ checksum ".pre-commit-config.yaml" }}-{{ checksum "tox.ini" }}
+          key: sceptre-v1-{{ checksum ".pre-commit-config.yaml" }}
           paths:
-            - '~/.cache/pypoetry'
-            - '~/.cache/pre-commit'
-            - '.tox'
+            - ~/.cache/pre-commit
 
   unit-tests:
-    docker:
-      - image: sceptreorg/sceptre-circleci:2.1.0
+    parameters:
+      python:
+        type: string
+    executor:
+      name: python/default
+      tag: <<parameters.python>>
     steps:
       - checkout
       - restore_cache:
-          key: sceptre-v1-{{ checksum "pyproject.toml" }}-{{ checksum "poetry.lock" }}-{{ checksum ".pre-commit-config.yaml" }}-{{ checksum "tox.ini" }}
+          key: sceptre-v1-{{ checksum "poetry.lock" }}-{{ checksum "tox.ini" }}
+      # Update poetry for https://github.com/python-poetry/poetry/issues/7184
+      - run: poetry self update --no-ansi
+      - python/install-packages:
+          pkg-manager: poetry
+          args: --all-extras
       - run:
           name: Unit Tests
           command: |
```

Comment:
I attempt to explain it. Does it make sense? Can you make it more concise?

```suggestion
          # Normally a single tox invocation runs the tests on all available
          # Python versions.
          #
          # Now CircleCI runs parallel test pipelines, one for each supported
          # Python version.
          #
          # See `matrix.parameters.python` for the Python version values. Each
          # pipeline uses that value as its default Python version.
          #
          # There is one tox invocation for each pipeline. The `-e` option runs
          # tox on just the default Python for that pipeline.
          #
          # The container may make more Python versions available, and we don't
          # want tox to run the tests on those because they may cause false
          # failures.
          #
          # For example, if the container for Python 3.12 would use Python 3.12
          # by default. It make also have an earlier Python version installed
          # such as 3.10 that was used for bootstrapping the container image.
          # Configured this way, tox ignores the version 3.10 in the pipeline
          # for version 3.12.
          command: |
```